### PR TITLE
Add subclassing pattern/example

### PR DESCRIPTION
**Pull request by antsar**
Wrote up a basic example of subclassing Flask to set up an altered Jinja2 environment. This should fix issue #221

### CODE REVIEW

File: docs/patterns/subclassing.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+The :class:`~flask.Flask` class is designed for subclassing.
+
+One reason to subclass would be customizing the Jinja2 :class:`~jinja2.Environment`. For example, to add a new global template variable::
+
+    from flask import Flask
+    from datetime import datetime
+
+    class MyFlask(Flask):
+        """ Flask with more global template vars """
```
Comment (ThiefMaster) [inline docs/patterns/subclassing.rst]: There shouldn't be surrounding whitespace

File: docs/patterns/subclassing.rst [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+        def create_jinja_environment(self):
+            """ Initialize my custom Jinja environment. """
+            jinja_env = super(MyFlask, self).create_jinja_environment(self)
+            jinja_env.globals.update(
+                current_time = datetime.datetime.now()
```
Comment (ThiefMaster) [inline docs/patterns/subclassing.rst]: kwargs don't have space around the `=` and this example doesn't seem particularly useful; especially considering that this probably runs once at setup time and there are already proper APIs in Flask to add new template globals, filters or tests without subclassing.

**Comment by antsar** [comment]
Thanks for the quick feedback. Any suggestions for a better subclassing use-case?

File: docs/patterns/subclassing.rst (hunk 2 above)
Comment (antsar) [inline docs/patterns/subclassing.rst]: Thanks for the quick feedback. Any suggestions for a better subclassing use-case?

Comment (ThiefMaster) [inline docs/patterns/subclassing.rst]: I'm not sure if there is any good reason to override this specific method.

**Pull request merged**